src/components/navbar.component.tsx

```diff
@@ -56,6 +65,43 @@ const Navbar: React.FC<NavbarProps> = ({ user, onLogout, allowedLocales }) => {
     };
   }, []);
 
+  let [searchBarIsVisible, setSearchBarIsVisible] = React.useState<any>(false);
+
+  let [closeSearchBar, setCloseSearchBar] = React.useState<any>(false);
+
+  let setSearchBar = () => {
+    setSearchBarIsVisible((searchBarIsVisible = true));
+    if (searchBarIsVisible === true) {
+      document
+        .getElementById("searchBar")
+        .classList.add(styles.searchOverlayOn);
+
+      document.getElementById("searchIcon").classList.add(styles.search);
```

Comment:
you can use `useRef` hook to access the DOM Elements.

Reply:
Sure , thanks had ran into a couple of issues with useRef , but let me look into  a work around.

Reply:
Describe your issues - we may be able to help you.

Reply:
@FlorianRappl the issue in this case was coordinating event coordinating event the three hooks  , i.e useEffects() , useState , and useRef  to make sure i add a listener to  the search component , alter the state , and fire the event . That's what took me back to the Vanilla way to try and fix this. It had robbed plenty of my time.

Reply:
Then one more thing i failed to achieve the transition time of the overlay effect was not achieved regardless of the different parameters that i provided.

Reply:
Well, I'm not sure if learning React is a robbery of time - especially if you want to contribute to React code - but okay.

I'd like to see code of what you tried - to understand where you went wrong.

In the case above you don't need any DOM interaction at all. You'd just need to change a state and reflect in the rendering to it (you can just do something like `[classA, classB, conditional && classC].filter(Boolean)].join(' ')` to derive the class name).

Reply:
+1 to what @FlorianRappl , you don't need `useRef` here, you just need to make the `className` conditional in the JSX block.